src/NHibernate/AdoNet/OracleDataClientBatchingBatcher.cs

```diff
@@ -125,21 +125,32 @@ protected override void DoExecuteBatch(IDbCommand ps)
 				// this value is not a part of the ADO.NET API.
 				// It's and ODP implementation, so it is being set by reflection
 				SetArrayBindCount(arraySize);
-				int rowsAffected;
 				try
 				{
-					rowsAffected = _currentBatch.ExecuteNonQuery();
+					int rowsAffected;
+					try
+					{
+						rowsAffected = _currentBatch.ExecuteNonQuery();
+					}
+					catch (DbException e)
+					{
+						// Message from DbException could be useful
+						var additionalMessage = string.IsNullOrEmpty(e.Message) ? "." : string.Concat(": ", e.Message);
```

Comment:
In MinimalSQLExceptionConverter and SQLStateConverter the actual exception will end up as an inner exception of GenericADOException. So, I think it's fine. Or, we can amend the PR to remove this change.